test/foundry/unit/NotifiableRewardAccumulator/StakingV2SetupWithAccumulator.t.sol

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {console} from "forge-std/Test.sol";
+import {Migrate} from "../../../../scripts/Migrate.s.sol";
+import {StakingV1Setup} from "../../utils/setup/StakingV1Setup.t.sol";
+import {RewardEscrowV2} from "../../../../contracts/RewardEscrowV2.sol";
+import {EscrowMigrator} from "../../../../contracts/EscrowMigrator.sol";
+import {StakingRewardsV2} from "../../../../contracts/StakingRewardsV2.sol";
+import {EarlyVestFeeDistributor} from "../../../../contracts/EarlyVestFeeDistributor.sol";
+import {NotifiableRewardAccumulator} from "../../../../contracts/NotifiableRewardAccumulator.sol";
+import {IRewardEscrowV2} from "../../../../contracts/interfaces/IRewardEscrowV2.sol";
+import "../../utils/Constants.t.sol";
+
+contract StakingV2Setup is StakingV1Setup {
```

Comment:
**Thought:** Not convinced this file/test contract is needed, instead just update `Migrate.s.sol` with the new upgrade pattern, and all the other tests will run on top of it.